### ISSUE TITLE
remove_duplicates.py division by zero

### ISSUE THREAD

**Issue by pnorman** [mapillary/mapillary_tools#101]
Off of an install in a clean virtualenv

```
$ python remove_duplicates.py -d 2 ../../20151028/01
Traceback (most recent call last):
  File "remove_duplicates.py", line 417, in <module>
    image_remover.do_magic()
  File "remove_duplicates.py", line 262, in do_magic
    capturetime, files = self._sort_file_list(files)
  File "remove_duplicates.py", line 251, in _sort_file_list
    capture_times = [self._read_capture_time(filepath) for filepath in file_list]
  File "remove_duplicates.py", line 244, in _read_capture_time
    reader = PILExifReader(filepath)
  File "/home/pnorman/osm/photos/mapillary_tools/python/lib/exif_pil.py", line 23, in __init__
    self._exif = self.get_exif_data(image)
  File "/home/pnorman/osm/photos/mapillary_tools/python/lib/exif_pil.py", line 31, in get_exif_data
    info = image._getexif()
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 391, in _getexif
    return _getexif(self)
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 425, in _getexif
    exif.update(info)
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 354, in __getitem__
    self[tag] = handler(self, data, self.legacy_api)  # check type
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 484, in load_rational
    for num, denom in zip(vals[::2], vals[1::2]))
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 484, in <genexpr>
    for num, denom in zip(vals[::2], vals[1::2]))
  File "/home/pnorman/osm/photos/ve/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 482, in <lambda>
    combine = lambda a, b: (a, b) if legacy_api else a / b
ZeroDivisionError: division by zero
```

**Comment by pnorman** [comment]
This leaves mapillary code with 

```python
from PIL import Image
image = Image.open(filepath)
image._getexif()
```

The first oddity is that the Pillow API docs don't describe a _getexif() function for images.

But the cause is that pillow exif is badly broken for images where digital zoom is not used: python-pillow/Pillow#1492.

saimn/sigal#172 shows that a bunch of exif stuff in Pillow 3 has changed, and their workaround was to use Pillow==2.9.0

**Comment by wiredfool** [comment]
For anyone who's interested in the exif handling in Pillow, Please test the new 3.1.0-rc1 release soon and give feedback. We've tried to fix the issues referenced here.

**Comment by pnorman** [comment]
I can confirm pillow 3.1.0 fixes it. Not quite sure how to properly specify the version to allow 2.9 and 3.1 but not 3.0

**Comment by ybkuang** [comment]
This looks like a problem related to PIL. We will be replacing the exif parsing with exifread across the lib in the future. Closing this for now.

**Comment by pnorman** [comment]
> This looks like a problem related to PIL. We will be replacing the exif parsing with exifread across the lib in the future. Closing this for now.

It's already closed by #102
